packages/config/src/integrations.ts

```diff
@@ -0,0 +1,65 @@
+import { getAvailableIntegrations } from './api/integrations.js'
+import { IntegrationResponse } from './types/api.js'
+import { Integration } from './types/integrations.js'
+
+type MergeIntegrationsOpts = {
+  configIntegrations?: { name: string; dev?: { path: string; force_run_in_build?: boolean } }[]
+  apiIntegrations: IntegrationResponse[]
+  context: string
+}
+
+export const mergeIntegrations = async function ({
+  configIntegrations = [],
+  apiIntegrations,
+  context,
+}: MergeIntegrationsOpts): Promise<Integration[]> {
+  const availableIntegrations = await getAvailableIntegrations()
+
+  // Include all API integrations, unless they have a `dev` property and we are in the `dev` context
+  const resolvedApiIntegrations = apiIntegrations.filter(
```

Comment:
IMO this type of nesting with callbacks 3 levels deep makes the code super hard to read. I think adding some intermediate variables with descriptive names would make the lives of our future selves much easier.

Reply:
Yeah it's pretty grim, will mark it down to refactor this properly in the future - we need to revisit relatively soon as we're touching on some further logic around managing integrations in config